src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/custom.py

```diff
@@ -124,64 +125,74 @@ def acr_show(registry_name, resource_group_name=None):
 
     return client.get_properties(resource_group_name, registry_name)
 
-def acr_update(registry_name, #pylint: disable=too-many-arguments
-               resource_group_name=None,
-               storage_account_name=None,
-               tags=None,
-               enable_admin=False,
-               disable_admin=False):
-    '''Update a container registry.
-    :param str registry_name: The name of container registry
-    :param str resource_group_name: The name of resource group
-    :param str storage_account_name: The name of storage account
-    :param dict tags: The set of tags
-    :param bool enable_admin: Enable admin user
-    :param bool disable_admin: Disable admin user
-    '''
+def acr_update_set(registry_name,
+                   resource_group_name=None,
+                   parameters=None):
     registry, resource_group_name = get_registry_by_name(registry_name, resource_group_name)
 
-    # Set admin_user_enabled
-    admin_user_enabled = None
-    if disable_admin:
-        admin_user_enabled = False
-    if enable_admin:
-        admin_user_enabled = True
-
-    # Set tags
-    newTags = registry.tags #pylint: disable=no-member
```

Comment:
If you do choose to keep the convenience arguments in (--tags) you can alias it with the global tags_type from `azure.cli.core.commands.parameters` and it shoud handle the conversion of key=value string pairs to dictionary for you. So you would not need any custom logic to do this.